libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -32,11 +32,36 @@
 #include <libsolutil/StringUtils.h>
 #include <libsolidity/ast/TypeProvider.h>
 
+#include <range/v3/algorithm/all_of.hpp>
+
 using namespace solidity;
 using namespace solidity::util;
 using namespace solidity::frontend;
 using namespace std::string_literals;
 
+namespace
+{
+
+std::optional<size_t> staticEncodingSize(std::vector<Type const*> const& _parameterTypes)
+{
+	size_t encodedSize = 0;
+	for (auto* type: _parameterTypes)
+	{
+		if (dynamic_cast<StringLiteralType const*>(type))
+			// Workaround for StringLiteralType, which will fail an assert on the
+			// type->calldataHeadSize() call; thus, assign string memory type (array)
+			// to type, which was and is explicitly provided to revertWithError where
+			// applicable anyway.
+			type = TypeProvider::stringMemory();
```

Comment:
Ah, good point (latter) Kamil; I wouldn't necessarily say it was a bug, because the string literal would then end up being encoded as a bytes array, just like `stringMemory()`, but this way is definitely better and more elegant as it allows us to remove the `StringLiteral` check/hack in the `staticEncodingSize()` function. This then makes the first comment (i.e. swapping the `if`s) unnecessary :)